firmware/src/rdm_handler.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * rdm_handler.h
+ * Copyright (C) 2015 Simon Newton
+ */
+
+/**
+ * @defgroup rdm_handler RDM Request Handler
+ * @brief The handler for RDM requests.
+ *
+ * This represents the entry point for RDM request handling. The handler
+ * dispaches the request to the appropriate model.
+ *
+ * @addtogroup rdm_handler
+ * @{
+ * @file rdm_handler.h
+ * @brief The handler for RDM requests.
+ */
+
+#ifndef FIRMWARE_SRC_RDM_HANDLER_H_
+#define FIRMWARE_SRC_RDM_HANDLER_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "iovec.h"
+#include "rdm_frame.h"
+#include "rdm_model.h"
+#include "rdm_responder.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief The callback used to send RDM responses.
+ * @param include_break true if a break should be used for the response
+ * @param data the raw data to send in the response
+ * @param iov_count the number of IOVec that make up the response
+ */
+typedef void (*RDMHandlerSendCallback)(bool include_break,
+                                       const IOVec* data,
+                                       unsigned int iov_count);
+
+/**
+ * @brief The settings to use for the RDM Handler.
+ */
+typedef struct {
+  uint8_t uid[UID_LENGTH];  //!< The responder's UID.
+  uint16_t default_model;  //!< The default model to use
```

Comment:
Probably, but I need to make sure it's extensible by others, possibly outside our codebase. I may change it later to an enum.